File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    answer = []
    sum_nums = 0
    l = r = 0
    min_len = 10000002
    
    while l < len(sequence):
        while r < len(sequence):
            if sum_nums >= k:
                break
            sum_nums += sequence[r]
            r += 1
        
        if sum_nums == k:
            if r - l < min_len:
                answer = [l, r - 1]
                min_len = r - l
        
        sum_nums -= sequence[l]
        l += 1

    return answer

File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int n = sequence.length;
        int sum = 0;
        int r = 0;
        int minLength = n + 1;
        
        int left = 0;
        int right = 0;
        
        for(int l = 0; l < n; l++) {
            while(r < n && sum < k) {
                sum += sequence[r];
                r++;
            }
            
            if(sum == k) {
                if(r - l < minLength) {
                    left = l;
                    right = r - 1;
                    minLength = r - l;
                }
            }
            sum -= sequence[l];
        }
        
        int[] answer = {left, right};
        return answer;
    }
}